### PR TITLE
nonhuman genetic disease pattern

### PR DESCRIPTION
**Pull request by sabrinatoro**
Addresses #6085

I also updated the existing patterns for genetic diseases in human to point to the specific "human disease" and not general "disease"

### CODE REVIEW

File: src/patterns/dosdp-patterns/nonhuman_disease_series_by_gene.yaml [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+  - gene
+
+def:
+  text: Any %s in which the cause of the disease is a variation in the %s gene.
+  vars:
+  - disease
+  - gene
+
+equivalentTo:
+  text: '%s and ''has material basis in germline mutation in'' some %s'
```
Comment (sabrinatoro) [inline src/patterns/dosdp-patterns/nonhuman_disease_series_by_gene.yaml:56]: @cmungall and @matentzn 
I did not add the taxon here because the taxon is included in the gene record. 
In your opinion, do you think we should add taxon information here? Is there a reason to add this redundancy? 
Thank you!

**Comment by matentzn** [comment]
Best add to Tech call agenda

**Comment by twhetzel** [comment]
Added, but the next Tech call is not until Friday, June 21

File: src/patterns/dosdp-patterns/nonhuman_disease_series_by_gene.yaml (hunk 1 above)
Comment (cmungall) [reply src/patterns/dosdp-patterns/nonhuman_disease_series_by_gene.yaml:56]: You need either

1. a property chain `'has material basis in germline mutation in' o 'in taxon' -> 'in taxon'`, in RO
2. the ensure the genus is already species-specific. Presumably this is not always the case as some PS analogs will be cross-mammal, cross-vertebrate, etc
3. just make a var for taxon even if it is duplicative with either/both genus/differentia

But going back to your Q from slack about infectious disease classification:

1. I would in general try and keep the ladders parallel. If some order/class/etc level virus is a useful grouping for H, it should also be for NH? But I appreciate this may not always be the case, esp for agents that never infect human, in which case
2. create a `non_human_disease_by_infectious_agent` DP
     - this would only use NH terms as genus
     - the taxon of the host would need to go in as a variable
     - classification would follow NCBITaxonomy
     - we can worry about making the xp-analog relations complete via rules later

If you do go with 2 it points to doing the same this for this PR, e.g. including a taxon var

**Comment by sabrinatoro** [comment]
@katiermullen @matentzn @cmungall 
I updated the pattern to include the taxon information (decided to go the redundancy route).
Could one of you please review and approve? Thank you!

File: src/patterns/dosdp-patterns/nonhuman_disease_series_by_gene.yaml [hunk 2]
```diff
@@ -22,38 +24,45 @@ annotationProperties:
 vars:
   disease: '''disease'''
   gene: '''gene'''
+  taxon: "'taxon'"
```
Comment (katiermullen) [inline src/patterns/dosdp-patterns/nonhuman_disease_series_by_gene.yaml:27]: For my education - is this supposed to be '''taxon''' and not "'taxon'"?

**Review by katiermullen — APPROVED** [review]
@sabrinatoro I have one tiny comment. Can you please take a look? Otherwise, it looks great to me and can be merged.

File: src/patterns/dosdp-patterns/nonhuman_disease_series_by_gene.yaml [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+def:
+  text: Any %s, occuring in %s, in which the cause of the disease is a variation in the %s gene.
+  vars:
+  - disease
+  - taxon
+  - gene
+
+equivalentTo:
+  text: '%s and (''has material basis in germline mutation in'' some %s) and (''in taxon'' some %s)'
```
Comment (matentzn) [inline src/patterns/dosdp-patterns/nonhuman_disease_series_by_gene.yaml:64]: Note here that there are two ways you can model this:

1. you can push the taxon into the gene. assuming you always use hgnc identifiers for humans and ncbigenes for animals, and you always use the correct gene id pertaining to a taxon specific gene, this is enough, no need for recording the taxon specifically to distinguish from human disease
2. however, you likely want the taxon to separate grouping classes "dog disease", which means having the taxon can be helpul (unless the genes in our ingest have them already!). You might also need this for non-gene non-human-animal diseases, so maybe just curate redundantly. If you do, however leave it like you have it, not that _technically_, you may want to add the human taxon to the human diseases. 

Long story short: its probably best to leave it has you have it, just adding some considerations in for future decision making.

**Review by matentzn — APPROVED** [review]
LGTM!

File: src/patterns/dosdp-patterns/nonhuman_disease_series_by_gene.yaml [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+
+description: 'This pattern is for non-human diseases that are defined based on a variation in one gene, such as many genetic diseases from the Online Mendelian Inheritance in Animals (OMIA).  Examples: TBD '
+
+contributors:
+- https://orcid.org/0000-0002-4142-7153
+- https://orcid.org/0000-0002-5002-8648
+
+classes:
+  disease: MONDO:0005583
+  gene: SO:0001217
```
[the remark this answers is not in the document]
Comment (sabrinatoro) [reply src/patterns/dosdp-patterns/nonhuman_disease_series_by_gene.yaml:13]: Good call. I will also update the human pattern

File: src/patterns/dosdp-patterns/nonhuman_disease_series_by_gene.yaml (hunk 3 above)
Comment (sabrinatoro) [reply src/patterns/dosdp-patterns/nonhuman_disease_series_by_gene.yaml:64]: After a lot of discussion, we decided to keep the redundancy. But I agree, we might decide to remove this redundancy at a later time
